walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,77 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+import java.util.List;
+import java.util.ArrayList;
+import com.google.sps.TimeRange;
+import java.util.Comparator;
+import com.google.sps.Event;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        System.out.println("THen end of the day is: " + TimeRange.END_OF_DAY);
+        long duration = request.getDuration();
+        Collection<String> optionalAttendees = request.getOptionalAttendees();
+        Collection<String> mandatoryAttendees = request.getAttendees();
+        Collection<String> allAttendees = new ArrayList<String>();
```

Comment:
Do you need this list? Its just a concatenation of your previous lists, and you might be able to get rid of having to call returnMeetingTimes multiple times.
This probably depends on the expected behavior you are going for though.

Reply:
I need one list with mandatory and optional and one with just optional. Since both getAttendees and getOptionalAttendees return unmodifiable collections, I couldn't just add optional onto mandatory which is why i made a new list.